tests/scripts/spack_manifest/test_getter.py

```diff
@@ -0,0 +1,480 @@
+        manifest = {
+            "spack": {
+                "packages": {
+                    "package1": {
+                        "require": [
+                            "@git.1.0.0",
+                            "+debug",
+                            "%compiler@2.0.0"
+                        ]
+                    }
+                }
+            }
+        }
+
+        packages_getter = Packages(manifest)
+        requirements = packages_getter.get_package_requirements("package1")
+
+        assert requirements == ["@git.1.0.0", "+debug","%compiler@2.0.0"], "Package requirements should be correctly retrieved."
+
+    def test_get_package_requirements__invalid_no_requirements(self):
+        manifest = {
+            "spack": {
+                "packages": {
+                    "package1": {}
+                }
+            }
+        }
+
+        packages_getter = Packages(manifest)
+
+        assert packages_getter.get_package_requirements("package1") == [], "Package with no requirements should return an empty list."
+
+    def test_get_package_ref_requirement__valid(self):
+        manifest = {
+            "spack": {
+                "packages": {
+                    "package1": {"require": ["@git.1.0.0"]},
+                    "package2": {"require": ["@git.2.0.0"]},
+                }
+            }
+        }
+
+        packages_getter = Packages(manifest)
+        ref = packages_getter.get_package_ref_requirement("package1")
+
+        assert ref == "1.0.0", "Package ref requirement should be correctly retrieved."
+
+    def test_get_package_ref_requirement__invalid_no_package(self):
+        manifest = {
+            "spack": {
+                "packages": {
+                    "package1": {"require": ["@git.1.0.0"]},
+                }
+            }
+        }
+
+        packages_getter = Packages(manifest)
+        with pytest.raises(NoSectionComponentError):
+            packages_getter.get_package_ref_requirement("nonexistent_package")
+
+####################
+
+class TestProjectionsGetter:
+    def test_from_file__invalid(self):
+        manifest_path = "tests/scripts/spack_manifest/injection/inputs/invalid_manifest.yaml"
+
+        with pytest.raises(OSError):
+            Projections.from_file(manifest_path)
+
+    def test__get_defined_projections__valid(self):
+        manifest = {
+            "spack": {
+                "modules": {
+                    "default": {
+                        "tcl": {
+                            "projections": {
+                                "package1": "package1/1.0.0",
+                                "package2": "package2/2.0.0",
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        expected = {"package1": "package1/1.0.0", "package2": "package2/2.0.0"}
+
+        projections_getter = Projections(manifest)
+        result = projections_getter.get()
```

Comment:
Just from a lines of code point of view could collapse with no loss of clarity

```suggestion
        result = Projections(manifest).get()
```